js/script.js

```diff
@@ -0,0 +1,199 @@
+//         released: "25 May 1977",
+//         actors: ["Mark Hamill", "Harrison Ford", "Carrie Fisher"],
+//         plot: "Luke Skywalker joins forces with a Jedi Knight, a cocky pilot, a Wookiee and two droids to save the galaxy from the Empire's world-destroying battle station, while also attempting to rescue Princess Leia from the mysterious Darth Vad",
+   
+let actorsInfo = mainArray
+    .reduce((total, {title, actors}) => {
+    return [...total, {title, actors}]
+    }, [])
+    .reduce((total, currObj, i, arr)=>{
+        let actorsArr = arr.map(el => el.actors).flat(Infinity).filter((el, i, arr) => (arr.indexOf(el) === i));
+        for(let val of actorsArr){
+            if(currObj.actors.includes(val)){
+                if (total.hasOwnProperty(val)) {
+                    return {
+                      ...total,
+                      [val]: [...total[val], currObj.title],
+                    };
+                  }  
+                   return {
+                    ...total, [val]: [currObj.title]
+                   }
+            }  
+        }
+    }, {}) 
+;
+
+
+    console.log(actorsInfo)
+// 6. Создать массив авторов (поле writer) без повторений.
+
+writerArray = mainArray
+     .map(el => el.writer).flat(Infinity).filter((el, i, arr) => (arr.indexOf(el) === i));
+
+// 7. Создать функцию, которая бы принимала массив фильмов и строку. А результатом этой функции должен быть новый отфильтрованный массив, с фильмами, где строка входит в название фильма.
+    
+    function getFilmsTitles(filmArray, currStr){
+        return filmArray.filter(({title}) => title.indexOf(currStr) > -1)
+    
+    }
+    
+
+// 8. Создать функцию, которая бы принимала массив фильмов и число. А результатом этой функции должен быть отфильтрованный массив, с фильмами где число равно году выхода фильма.
+    
+    function getFilmsYear(filmArray, currYear){
+        return filmArray.reduce((total, {title, year}) => {
+            if(year === currYear){
+                return [ ...total,{ [title]: year } ]
+            }
+         return total;
+        }, [])
+    }
+    
+// 9. Создать функцию, которая бы принимала массив фильмов и строку. А результатом этой функции должен быть отфильтрованный массив, с фильмами где строка входит в название фильма или в его сюжет.
+
+    function checkFilmsTitlesOrPlot(filmArray, currStr){
+        return filmArray.filter(({title, plot}) => ((title.indexOf(currStr) > -1) || (plot.indexOf(currStr) > -1)))
+    }
+
+// 10. Создать функцию, которая бы принимала 3 параметра:  1)массив фильмов , 2) строка(название поля, например 'title') 
+// и строку/число(значение поля "Black Widow").
+// А результатом этой функции должен быть отфильтрованный массив, где параметры 2 и 3 равны в объекте фильма. 
+// Например: передаем `(films, 'title', 'Black Widow')` и на выходе получаем фильм с id=1
+// если передаем `(films, 'year', 2011)` , то получаем фильм с id=2
+
+    function getFilm(filmArray, prop, searchStr){
```

Comment:
1. should be arrow function
2. prop -> mb better to write -> fieldName
3. searchStr -> fieldValue